### PR TITLE
Upgrade to Curator 5.1

### PR DESCRIPTION
**Pull request by eolivelli**
Descriptions of the changes in this PR:

Curator 5.x support ZooKeeper 3.6.x.
This change is needed to able the upgrade of Pulsar to ZooKeeper 3.6.2 

### Changes

Upgrade Curator

### CODE REVIEW

**Comment by eolivelli** [comment]
rerun failure checks

File: stream/storage/impl/src/main/java/org/apache/bookkeeper/stream/storage/impl/cluster/ZkClusterMetadataStore.java [hunk 1]
```diff
@@ -58,7 +58,7 @@
     private final String zkClusterAssignmentPath;
 
     private final Map<Consumer<Void>, NodeCacheListener> assignmentDataConsumers;
-    private NodeCache assignmentDataCache;
+    private org.apache.curator.framework.recipes.cache.NodeCache assignmentDataCache;
```
Comment (Ghatage) [inline stream/storage/impl/src/main/java/org/apache/bookkeeper/stream/storage/impl/cluster/ZkClusterMetadataStore.java:61]: Sorry I think I might be missing something. Are there overloaded types of `NodeCache` in the new version?
Why are we giving the fully qualified class name?

Comment (eolivelli) [reply stream/storage/impl/src/main/java/org/apache/bookkeeper/stream/storage/impl/cluster/ZkClusterMetadataStore.java:61]: NodeCache as been deprecated, there is a new API, but I wanted to keep the patch as small as possible

Comment (Ghatage) [reply stream/storage/impl/src/main/java/org/apache/bookkeeper/stream/storage/impl/cluster/ZkClusterMetadataStore.java:61]: I see, thanks

**Review by Ghatage — APPROVED** [review]
LGTM

**Pull request merged**